pagefaultgames/pokerogue
# [Move] Heal Block

## What are the changes?
This implements Heal Block and provides a test for it. 

## Why am I doing these changes?
I think it would be nice for it to be implemented. 

## What did change?
- HealBlockTag (Duration: 5 Turns)
- PokemonHealPhase now accounts for HealBlockTag 
- Not implemented yet: moves restricted by heal block 
- Not implemented yet: Tests 
- Lines for Localization 

### Screenshots/Videos
https://github.com/user-attachments/assets/08b71f5d-3325-437d-b4d4-975b7f803973

## How to test the changes?
Use Heal Block during a Pokemon battle and see if the affected pokemon can heal itself. 

## Checklist
- [] Have I provided a clear explanation of the changes?
    - [] Are all unit tests still passing? (`npm run test`)
